src/gui/SubWindow.cpp

```diff
@@ -26,42 +26,249 @@
 
 #include "SubWindow.h"
 
+#include <QMdiArea>
 #include <QMoveEvent>
 #include <QResizeEvent>
-#include <QWidget>
+
+#include "embed.h"
 
 
 SubWindow::SubWindow(QWidget *parent, Qt::WindowFlags windowFlags)
-  : QMdiSubWindow(parent, windowFlags)
+	: QMdiSubWindow(parent, windowFlags), m_buttonSize (17,17)
 {
 	// initialize the tracked geometry to whatever Qt thinks the normal geometry currently is.
 	// this should always work, since QMdiSubWindows will not start as maximized
 	m_trackedNormalGeom = normalGeometry();
+
+	// inits the colors
+	m_activeColor = Qt::SolidPattern;
+	m_textShadowColor = Qt::black;
+	m_borderColor = Qt::black;
+
+	//close, minimize, maximize and restore(after minimize) buttons
+	m_closeBtn = new QPushButton( embed::getIconPixmap( "close" ), QString::null, this );
+	m_closeBtn->resize(m_buttonSize);
+	m_closeBtn->setFocusPolicy(Qt::NoFocus);
+	m_closeBtn->setToolTip(tr("Close"));
+	connect(m_closeBtn, SIGNAL(clicked(bool)), this, SLOT(close()));
+
+	m_maximizeBtn = new QPushButton( embed::getIconPixmap( "maximize" ), QString::null, this );
+	m_maximizeBtn->resize(m_buttonSize);
+	m_maximizeBtn->setFocusPolicy(Qt::NoFocus);
+	m_maximizeBtn->setToolTip(tr("Maximize"));
+	connect(m_maximizeBtn, SIGNAL(clicked(bool)), this, SLOT(showMaximized()));
+
+	m_minimizeBtn = new QPushButton( embed::getIconPixmap( "minimize" ), QString::null, this );
+	m_minimizeBtn->resize(m_buttonSize);
+	m_minimizeBtn->setFocusPolicy(Qt::NoFocus);
+	m_minimizeBtn->setToolTip(tr("Minimize"));
+	connect(m_minimizeBtn, SIGNAL(clicked(bool)), this, SLOT(showMinimized()));
+
+	m_restoreBtn = new QPushButton( embed::getIconPixmap( "restore" ), QString::null, this );
+	m_restoreBtn->resize(m_buttonSize);
+	m_restoreBtn->setFocusPolicy(Qt::NoFocus);
+	m_restoreBtn->setToolTip(tr("Restore"));
+	connect(m_restoreBtn, SIGNAL(clicked(bool)), this, SLOT(showNormal()));
+
+	// QLabel for window title and shadow effect
+	m_shadow = new QGraphicsDropShadowEffect();
+	m_shadow->setColor(m_textShadowColor);
+	m_shadow->setBlurRadius(3);
```

Comment:
I'd suggest this blur gets removed, since all the other drop shadows in LMMS are not blurred, so this is inconsistent with the rest of the codebase.